### DISCUSSION TITLE
`quarto install tool` or `quarto tools install`?

### DISCUSSION THREAD

**Discussion by mcanouil** [quarto-dev/quarto-cli#1437] [Q&A] [answered]
While in the documentation website, it is mostly `quarto tools install`, in Quarto CLI, some messages highlight `quarto install tool`.
What is the proper/preferred way (i.e., both work)?

To note, the uninstall subcommand is only available through `quarto tools uninstall`.

I spotted the following places where `quarto install tool` is "used":

- https://github.com/quarto-dev/quarto-cli/blob/763b0ed7f060b206ef9b3a247612126b5f2f1f2a/.github/workflows/test-smokes.yml#L67-L69
- https://github.com/quarto-dev/quarto-cli/blob/763b0ed7f060b206ef9b3a247612126b5f2f1f2a/src/command/install/cmd.ts#L46-L57
- https://github.com/quarto-dev/quarto-cli/blob/763b0ed7f060b206ef9b3a247612126b5f2f1f2a/src/command/render/latexmk/latex.ts#L214-L216
- https://github.com/quarto-dev/quarto-cli/blob/763b0ed7f060b206ef9b3a247612126b5f2f1f2a/src/tools/tools.ts#L141-L143
- https://github.com/quarto-dev/quarto-cli/blob/763b0ed7f060b206ef9b3a247612126b5f2f1f2a/src/tools/tools.ts#L168-L170

**Comment by jjallaire** [comment]
I think the docs should use `quarto install tool`. @dragonstyle Could you update the docs to be consistent in this usage?

**Reply by mcanouil** [reply]
This would mean to install, for example, `tinytex` with `quarto install tool tinytex` (I believe this command is more clear) and to uninstall it with `quarto tools uninstall tinytex`.  
Not sure, the uninstall command is straightforward though.  
Maybe if a similar subcommand as `install` existed, that would ease things up, i.e., `quarto uninstall <type> [target]`.  
Anyway `tools` is not even listed in `quarto --help`, so definitely not the command to keep.

<img width="738" alt="image" src="https://user-images.githubusercontent.com/8896044/179416178-3c4ab3fe-3069-4eed-ad9b-dc252b812411.png">

**Comment by jjallaire** [comment]
Yes `tools` is deprecated so we will be removing it from the docs.

**Comment by dragonstyle** [comment, marked as answer]
I've corrected the quarto.org documentation (deploying now). You can use:

`quarto remove tool tinytex` 

to handle uninstallation and 

`quarto update tool tinytex` 

to update to the latest version.

`quarto list tools`

will get you a summary of the tools.

We're in the process of improving these commands - I think making update/remove/list visible in help may have slipped through the cracks - we'll get that corrected when we can.

**Reply by mcanouil** [reply]
Perfect `remove` will do 😉

PS: Which posts deserves to be marked as the answer? 😏
